cmd/state-svc/internal/resolver/resolver.go

```diff
@@ -50,11 +50,22 @@ func (r *Resolver) AvailableUpdate(ctx context.Context) (*graph.AvailableUpdate,
 		return nil, errs.Wrap(err, "Failed to check for update")
 	}
 
-	av := graph.AvailableUpdate{
-		Available: update != nil,
+	available := true
+	if update == nil {
+		available = false
+		update = &updater.AvailableUpdate{}
```

Comment:
If no update is available then the response should be empty/null.

Reply:
This is done. I strongly recommend against returning `nil, nil`. It is not common, and can lead to mistakes. It is common for there to be explicit verification of data through errors or "ok" values (the "Available" field was the ok value in this case) with the most major exception being types which are commonly used as `nil` like slices.